packages/chain-bridge/src/adapters/StargateAdapter.sol

```diff
@@ -0,0 +1,979 @@
+        address, // keeper
+        BridgeTypes.AdapterParams calldata
+    ) external payable {
+        revert OperationNotSupported();
+    }
+
+    /// @inheritdoc ISendAdapter
+    function sendMessage(
+        bytes32,
+        uint16,
+        address,
+        bytes calldata,
+        address, // keeper
+        BridgeTypes.AdapterParams calldata
+    ) external payable {
+        revert OperationNotSupported();
+    }
+
+    /*//////////////////////////////////////////////////////////////
+                          HELPER FUNCTIONS
+    //////////////////////////////////////////////////////////////*/
+
+    /**
+     * @notice Handles composed messages from LayerZero after Stargate token delivery
+     * @dev Called by LayerZero endpoint after tokens are delivered via Stargate V2
+     * @param _from The originating OApp (should be destination Stargate contract)
+     * @param _message OFT-encoded compose message from Stargate
+     */
+    function lzCompose(
+        address _from,
+        bytes32,
+        bytes calldata _message,
+        address,
+        bytes calldata
+    ) external payable override {
+        // Verify caller is LayerZero endpoint
+        if (msg.sender != lzEndpoint) {
+            revert Unauthorized();
+        }
+
+        // Extract the amount and compose message from OFT encoding
+        uint256 amountLD = OFTComposeMsgCodec.amountLD(_message);
+        bytes memory composeMsg = OFTComposeMsgCodec.composeMsg(_message);
+
+        // Decode compose message and handle the rest
+        _handleComposedMessage(_from, amountLD, composeMsg);
+    }
+
+    /**
+     * @dev Internal function to handle the composed message logic - SIMPLIFIED
+     */
+    function _handleComposedMessage(
+        address _from,
+        uint256 amountLD,
+        bytes memory composeMsg
+    ) internal {
+        // Decode the compose message
+        (
+            address recipient,
+            address sourceAsset,
+            ,
+            uint256 sourceChainId,
+            bytes32 operationId,
+            address originator
+        ) = abi.decode(
+                composeMsg,
+                (address, address, uint256, uint256, bytes32, address)
+            );
+
+        // Get destination asset - SIMPLIFIED: fail fast if we can't get it
+        address destinationAsset = IStargate(_from).token();
+
+        // Basic validation - fail fast
+        if (
+            destinationAsset == address(0) ||
+            recipient == address(0) ||
+            amountLD == 0
+        ) {
+            revert InvalidParams();
+        }
+
+        // Check adapter balance
+        uint256 adapterBalance = IERC20(destinationAsset).balanceOf(
+            address(this)
+        );
+        if (adapterBalance < amountLD) {
+            revert InsufficientBalance();
+        }
+
+        // Determine if this is a deposit or withdrawal based on recipient type
+        bool isDeposit = _isFleetProxy(recipient);
```

Comment:
the withdrawals are initiated by the `FleetProxy` and are not requested from source chain by the `CrossChainArks` - that seems unnecessary